hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -96,7 +101,8 @@ public void initialize() {
     // this operation returns the value as Data, except when it's retrieved from Near Cache with in-memory-format OBJECT
     @Override
     @SuppressWarnings("unchecked")
-    protected V getInternal(Data key) {
+    protected V getInternal(Object key) {
+        key = serializeKeys ? toDataWithStrategy(key) : key;
```

Comment:
I know the code paths which lead to additional serializations, and it's "just" the normal proxy and the `NearCacheRecordStore` to get the partition ID to retrieve the `MetaDataContainer`. I have a prototype for a fix on top of all other PRs: https://github.com/hazelcast/hazelcast/pull/10617/commits/86ba6d67944fe874e90aea1f43204f1a8fb63e16